lib/internal/validators.js

```diff
@@ -567,6 +568,46 @@ const validateLinkHeaderValue = hideStackFrames((hints) => {
   );
 });
 
+// 1. Returns false for undefined and NaN
+// 2. Returns true for finite numbers
+// 3. Throws ERR_INVALID_ARG_TYPE for non-numbers
+// 4. Throws ERR_OUT_OF_RANGE for infinite numbers
+const validateFiniteNumber = hideStackFrames((number, name) => {
+  // Common case
+  if (number === undefined) {
+    return false;
+  }
+
+  if (NumberIsFinite(number)) {
+    return true; // Is a valid number
+  }
+
+  if (NumberIsNaN(number)) {
+    return false;
+  }
+
+  validateNumber(number, name);
+
+  // Infinite numbers
+  throw new ERR_OUT_OF_RANGE(name, 'a finite number', number);
+});
+
+// 1. Returns def for number when it's undefined or NaN
+// 2. Returns number for finite numbers >= lower and <= upper
+// 3. Throws ERR_INVALID_ARG_TYPE for non-numbers
+// 4. Throws ERR_OUT_OF_RANGE for infinite numbers or numbers > upper or < lower
+const checkRangesOrGetDefault = hideStackFrames(
```

Comment:
IMO the validators shouldn't do an "or get default" because thats more than just validation.

Secondly, validating a range is `validateNumber`, so I don't think a separate function is needed.

Reply:
Any change would result in server-major change.